dataactvalidator/validation_handlers/validationManager.py

```diff
@@ -486,15 +503,20 @@ def runCrossValidation(self, jobId, interfaces):
             failures = Validator.crossValidateSql(comboRules.all(),submissionId)
             # get error file name
             reportFilename = self.getFileName(interfaces.errorDb.getCrossReportName(submissionId, row.first_file_name, row.second_file_name))
+            warningReportFilename = self.getFileName(interfaces.errorDb.getCrossWarningReportName(submissionId, row.first_file_name, row.second_file_name))
 
             # loop through failures to create the error report
-            with self.getWriter(regionName, bucketName, reportFilename,
-                                    self.crossFileReportHeaders) as writer:
+            with self.getWriter(regionName, bucketName, reportFilename, self.crossFileReportHeaders) as writer, \
+                 self.getWriter(regionName, bucketName, warningReportFilename, self.crossFileReportHeaders) as warningWriter:
                 for failure in failures:
-                    writer.write(failure[0:7])
+                    if failure[9] == interfaces.validationDb.getRuleSeverityId("fatal"):
```

Comment:
The getRuleSeverityId function goes through getNameFromDict, which actually caches the dict and only does a single DB call to populate it.